spec/lib/cleanup/destroy_unused_providers_spec.rb

```diff
@@ -0,0 +1,131 @@
+require 'rails_helper'
+require './lib/cleanup/destroy_unused_providers'
+
+RSpec.describe DestroyUnusedProviders do
+  let(:stdout) { StringIO.new }
+  let(:stdin) { StringIO.new }
+  let(:service_provider) { create(:service_provider) }
+  let(:issuer) { service_provider.issuer }
+
+  subject(:destroy_unused_providers) { described_class.new([issuer], stdout:, stdin:) }
+
+  before do
+    allow(stdin).to receive(:puts)
```

Comment:
Ditto here, we can construct the StringIO with a specific string and let the code read from it:

```suggestion
    yes_stdin = StringIO.new('yes')
    no_stdin = StringIO.new('no')
```